ctapipe/containers.py

```diff
@@ -114,6 +114,11 @@ class HillasParametersContainer(Container):
     length = Field(nan * u.m, "standard deviation along the major-axis", unit=u.m)
     width = Field(nan * u.m, "standard spread along the minor-axis", unit=u.m)
     psi = Field(nan * u.deg, "rotation angle of ellipse", unit=u.deg)
+    psi_divergent = Field(
```

Comment:
Why is this field needed? How is this calculated? Where does it come from.

I think this needs a much better explanation.

Reply:
If this is something specific to the HillasReconstructor, it should be stored in a container there, not in the container that is created by `hillas_parameters`.

Reply:
**Why is this field needed? How is this calculated? Where does it come from.**

This is explained in the thesis of Thomas Gasparetto which is the author of the relevant parts of this class (section 7.1.4).

In synthesis:
In the case of divergent pointing, the value of psi used to estimate the core position cannot be the one from the original camera frame.
In that case, in fact, it is not possible to simply project the psi angles from the camera frame / titled frame to the ground frame because all the camera planes are pointing somewhere else.
The correction comes from the fact that after getting the reconstructed direction in the sky (which estimation is not influenced by the single pointings) the image axis is projected onto a camera frame which points to the common the array pointing (therefore the name "fake" - in a divergent simulation no telescope is pointing there).
In this way one can then proceed to go from this frame to Tilted frame and finally ground frame.

**If this is something specific to the HillasReconstructor, it should be stored in a container there**

For the reasons above, yes it is something specific to the HillasReconstructor, so I it is better to add it to the shower container under the name `psi_core`.

Reply:
Ah wait, now I remember why I preferred to put it in HillasParametersContainer:
- psi_divergent is derived hillas parameters, so a telescope-wise information
- moving it elsewhere requires to add shower information to the plotting methods of array display which currently work with hillas dictionaries (and time gradients)

Reply:
Could you add references to the thesis at the appropriate positions?

Reply:
Is [this](https://cta-observatory.github.io/ctapipe/development/style-guide.html#api-documentation-style) the current way to do this?

Reply:
I don't see any reference related things in the link.

You have to add an entry in `docs/bibliography.rst` and reference it using `[reference]_`

Reply:
I also preferred not to have this in HillasParametersContainer, bu After discussing this, I think it is useful but misnamed. It is not the "divergent" psi, but rather the psi computed in the TiltedGroundFrame or CameraFrame, whereas all other hillas parameters are calculated in the TelescopeFrame.  Both are needed: the ground psi is needed for the core recon, and the telescope one for the point of origin (now that that is done in the TelescopeFrame rather than the CameraFrame).  

An alternate  would be to compute and store hillas parameters *for each frame*, even though only psi is necessary (for this algorithm at least, not sure of others).   Or even better, to just implement a function that computes psi in the TiltedGroundFrame if you have it already computed in the TelescopeFrame (an angle coordinate transform)

Reply:
So, to summarize:
- `psi_divergent` will for sure be renamed more correctly in e.g. `psi_core` (it doesn't matter if Tilted or GroundFrame because when I record it I do it to the already corrected - if needed because of the divergent mode - value)
- it shouldn't stay in HillasParametersContainer mainly because it does not belong to the TelescopeFrame anymore, rather to a cartesian frame
- it is basically needed by 2 things: the core estimation and its plotting